pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1051,10 +1051,12 @@ public void deleteCursorFailed(ManagedLedgerException exception, Object ctx) {
 
     void removeSubscription(String subscriptionName) {
         PersistentSubscription sub = subscriptions.remove(subscriptionName);
-        // preserve accumulative stats form removed subscription
-        SubscriptionStatsImpl stats = sub.getStats(false, false, false);
-        bytesOutFromRemovedSubscriptions.add(stats.bytesOutCounter);
-        msgOutFromRemovedSubscriptions.add(stats.msgOutCounter);
+        if (sub != null) {
```

Comment:
Maybe we could record the null sub by a warning log.

like this, just a demo code
```
if (sub == null) {
  log.warn("The subscription {} for topic {} is not existing.", subscriptionName, topic).
  return;
}
```

Reply:
+1 PTAL